src/pages/Signup.tsx

```diff
@@ -1,24 +1,27 @@
-import { useContext, SyntheticEvent, useState } from "react";
-import { SupabaseContext } from "../context/SupabaseContext.tsx";
+import { SyntheticEvent, useState } from "react";
 import { Box, Button, TextField, Typography } from "@mui/material";
+import supabase from "../supabase";
+import logo from "../assets/logo-medium.svg";
 
 export function Signup() {
-  const { supabase } = useContext(SupabaseContext);
-
+  const [name, setName] = useState("");
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [success, setSuccess] = useState(false);
   const [signupError, setSignupError] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
 
   const handleSubmit = async (e: SyntheticEvent<HTMLFormElement>) => {
     e.preventDefault();
 
     setSuccess(false);
     setSignupError("");
+    setIsLoading(true);
 
     const { error } = await supabase.auth.signUp({
       email,
       password,
+      options: { data: { name }, emailRedirectTo: location.origin },
```

Comment:
This specifies the destination URL for users who click the link in the verification email. The link should now redirect users to the correct page.